### PR TITLE
fix: user registration mail always sent in english

### PR DESCRIPTION
**Pull request by shauke**
## PR Type

[x] Bugfix

## What Is the Current Behavior?

User registration email is always sent in english (en_US) even if the user triggers the registration from a different storefront locale and with a this different locale in the REST call too.
This results from the hardcoded 'en_US' as preferredLanguage in the registration form.

## What Is the New Behavior?

The currently selected locale is used as 'preferredLanguage' when sending the registration form. With this change the registration email is send in the locale the user triggers the registration from.

## Does this PR Introduce a Breaking Change?

[ ] Yes
[x] No

### CODE REVIEW

**Review by dhhyi — CHANGES_REQUESTED** [review]
Hmm, I think, if this field is required by the API and not actually displayed on the registration form, why not just add it before sending in the [UserService.createUser](https://github.com/intershop/intershop-pwa/blob/7e357aca9c63bc85f9591411069316ec3aefd034/src/app/core/services/user/user.service.ts#L89) method. You also have the AppFacade there, you can use `withLatestFrom` in the stream and you can use '??' to set the parameter if it is not yet set.

File: src/app/core/services/user/user.service.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ import { ApiService, AvailableOptions } from 'ish-core/services/api/api.service'
 interface CreatePrivateCustomerType extends CustomerData {
   address: Address;
   credentials: Credentials;
+  user?: User;
```
Comment (shauke) [inline src/app/core/services/user/user.service.ts:28]: Any idea how I could get rid of this not correct addition to satisfy this line?
`{ ...newCustomer, user: { ...newCustomer.user, preferredLanguage: currentLocale.lang ?? 'en_US' } }`
Does this change reflect the idea you suggested?

File: src/app/core/services/user/user.service.ts [hunk 2]
```diff
@@ -107,11 +108,16 @@ export class UserService {
...
-          .post<void>(AppFacade.getCustomerRestResource(body.customer.isBusinessCustomer, isAppTypeRest), newCustomer, {
-            captcha: pick(body, ['captcha', 'captchaAction']),
-          })
+          .post<void>(
+            AppFacade.getCustomerRestResource(body.customer.isBusinessCustomer, isAppTypeRest),
+            body.customer.isBusinessCustomer
+              ? { ...newCustomer, user: { ...newCustomer.user, preferredLanguage: currentLocale.lang ?? 'en_US' } }
+              : { ...newCustomer, preferredLanguage: currentLocale.lang ?? 'en_US' },
+            { captcha: pick(body, ['captcha', 'captchaAction']) }
+          )
```
Comment (shauke) [inline src/app/core/services/user/user.service.ts:120]: Does this change reflect the idea you suggested?

Comment (dhhyi) [reply src/app/core/services/user/user.service.ts:120]: not quite the way I imagined. Nevertheless: let's be realistic: How fancy can we _actually_ become?! I guess, there's an optimum. But forget about that and be *realistic*. Just one method, but at the *right place* (THIS IS IMPORTANT!!!!!!). I'm good with that 🥰 
https://www.youtube.com/watch?v=gLCduDJVksc

Comment (shauke) [reply src/app/core/services/user/user.service.ts]: Let's give it one more try with the `newCustomer$` stream. ;)

File: src/app/core/services/user/user.service.ts [hunk 3]
```diff
@@ -85,29 +85,36 @@ export class UserService {
...
-        type: 'SMBCustomer',
-        ...body.customer,
-        user: { ...body.user },
-        address: customerAddress,
-        credentials: body.credentials,
-      };
-    }
+
+    let newCustomer$: Observable<CreatePrivateCustomerType | CreateBusinessCustomerType>;
+    newCustomer$ = this.appFacade.currentLocale$.pipe(
```
Comment (dhhyi) [inline src/app/core/services/user/user.service.ts:90]: Looks good 👍

**Review by dhhyi — APPROVED** [review]

**Pull request merged**